tests/foreman/ui/test_hostcollection.py

```diff
@@ -95,9 +96,10 @@ def module_repos_collection_module_stream(module_org, module_lce):
 
 @pytest.fixture
 def vm_content_hosts(request, module_loc, module_repos_collection):
+    distro = module_repos_collection.distro
     clients = []
     for _ in range(2):
-        client = VirtualMachine(distro=module_repos_collection.distro)
+        client = VMBroker(nick=distro, host_classes={'host': ContentHost})
```

Comment:
So, this only gets you a `VMBroker` instance - not the host instance. `VMBroker` does not have `create` or `destroy` methods defined, so I don't believe this will work at all.

This needs to be converted to use the VMBroker context manager, which will call `checkout` for all the hosts defined by the VMBroker instance ( the count can be included, like the use below, instead of iterating).

The call to `request.addfinalizer` can be removed, as well as the call to `client.create`, as the context manager takes care of both of these things.

Reply:
OK, changes made, but still to be tested